day3.sql

```diff
@@ -0,0 +1,316 @@
+SELECT * FROM JOBS;
+SELECT * FROM JOB_HISTROY;
+SELECT * FROM LOCATIONS;
+SELECT * FROM REGIONS;
+SELECT * FROM COUNTRIES;
+
+
+-- Question 1
+-- Write a SQL query to find the total salary of employees who is in Tokyo
+-- excluding whose first name is Nancy.
+
+select sum(salary) as Total_Salary
+from employee e
+join departments d 
+on e.department_id = d.department_id
+join locations l 
+on l.location_id = d.location_id
+where lower(city) = 'seattle'
+and
+lower(e.first_name) <> 'nancy';
+
+-- Question 2
+-- Fetch all details of employees who has salary more than the avg salary
+-- by each department.
+select * from employee e join 
+(select emp.department_id ,avg(salary) as salary from employee emp group by emp.department_id) e1
+on (e.department_id = e1.department_id and  e.salary > e1.salary ) order by e.salary;
+
+-- Question 3
+-- Write a SQL query to find the number of employees and its location whose salary is greater than or equal to 7000 and less than 10000
+select loc.city, count(employee.employee_id) as employee_count
+from departments as dept 
+join locations as loc 
+on dept.location_id = loc.location_id 
+join employee 
+on employee.department_id = dept.department_id 
+where employee.salary >= 7000 and employee.salary < 10000 group by city ;
+
+-- Question 4
+-- Fetch maxsalary,minsalary and avgsalary by job and department. 
+-- Info: grouped by department id and job id ordered by department id
+-- and max salary.
+select max(employee.salary) as max_sal,min(employee.salary) as min_salary,avg(employee.salary) as average_salary,department_id,job_id
+from employee
+group by department_id,job_id
+order by (department_id,max_sal);
+
+-- Question 5
+-- Write a SQL query to find the total salary of employees whose country_id is ‘US’ excluding whose first name is Nancy
+select
+sum(employee.salary) as total_salary from employee 
+join departments dept
+on dept.department_id = employee.department_id 
+join locations as location
+on location.location_id = dept.location_id
+where
+country_id='US'
+and 
+not employee.first_name='Nancy';
+
+-- Question 6
+-- Fetch max salary,min salary and avg salary by job id and department id but only for folks who worked in more than one role(job) in a department.
+
+select j1.department_id, j1.job_id, MAX(e.salary) as max_sal, MIN(e.salary) as min_sal, AVG(e.salary) as avg_sal 
+from job_history j1 
+join job_history j2  
+on j1.employee_id = j2.employee_id and j1.department_id = j2.department_id and j1.job_id != j2.job_id 
+join employee e 
+on j1.employee_id = e.employee_id 
+group by j1.job_id,j1.department_id;
+
+-- Question 7
+-- Display the employee count in each department and also in the same result.
+-- Info: * the total employee count categorized as "Total"
+-- • the null department count categorized as "-" *
+-- SELECT COALESCE(TO_VARCHAR(e.department_id), '-') as department_id,
+--        COUNT(*) as employee_count 
+-- FROM  employee e
+-- GROUP BY e.department_id;
```

Comment:
7 . add 
union 
![image](https://user-images.githubusercontent.com/129886415/231713219-4204dff0-a28e-4be1-acf3-adf54c935bcb.png)


This is an alternate for COALESCE

Reply:
Ok I will Update Ka